### ISSUE TITLE
Question - usage of objectWillSend within Published property on ObservableObject

### ISSUE THREAD

**Issue by rstrahl** [carekit-apple/CareKit#608]
Hi there.  In `OCKSimpleTaskController` there appears to be usage of `objectWillSend` despite the property being marked `Published`.  (refer to [main:OCKSimpleTaskController.swift](https://github.com/carekit-apple/CareKit/blob/5c3b14a4c2076e00c235c0357c490493908d1eb8/CareKit/CareKit/Shared/Task/Controller/OCKSimpleTaskController.swift#L38))

For convenience:
```swift
    @Published public private(set) var viewModel: SimpleTaskViewModel? {
        willSet { objectWillChange.send() }
    }
```

This seems superfluous (would it cause duplicate notifications?), but I may be missing something - is this intentional and serve a purpose? 

(If its incorrect/superfluous, I'd be more than happy to push a correction for review.)

Thanks - love the framework - looking forward to contributing!

**Comment by gavirawson-apple** [comment]
Great question! At the time the code was written, if a subclass of an `ObservableObject` has published properties, modifying their values will not automatically publish changes through the `objectWillChangePublisher`. The `OCKSimpleTaskController` is a subclass of the observable  `OCKTaskController`, so we hooked up the logic explicitly. 

It's all very possible that behavior has changed. If you find that the `willSet` block is no longer needed, we'd love to merge in a pull request and delete some code!

**Comment by rstrahl** [comment]
Ah, interesting - thanks for the clarification! I’ll run some tests, confirm whether that behaviour persists, and follow up here. Thanks!

**Comment by rstrahl** [comment]
I ran a simple test and confirmed the `Published` property works without the `willSet` block (Xcode 12.15.1 w/iOS 14.3).

Some further digging and found that this was resolved in iOS 14.5 ([release notes](https://developer.apple.com/documentation/ios-ipados-release-notes/ios-ipados-14_5-release-notes)):
> Using Published in a subclass of a type conforming to ObservableObject now correctly publishes changes. (71816443)

The 14.5 release was in April - fairly recent - whats the deprecation guidelines for CareKit?  (e.g. I could provide a PR that documents this for future cleanup in the affected `*TaskController` classes.)
